lib/miq_expression.rb

```diff
@@ -524,7 +524,22 @@ def self._to_ruby(exp, context_type, tz)
       operands = operands2rubyvalue(operator, exp[operator], context_type)
       clause = operands.join(" #{normalize_ruby_operator(operator)} ")
     when "contains"
-      operands = operands2rubyvalue(operator, exp[operator], context_type)
+      exp[operator]["tag"] ||= exp[operator]["field"]
+      operands = if context_type != "hash"
```

Comment:
darn.

I did like the consistency/brevity of all of these `case`/`when` statements.

is it possible to extract at least some of this into a method?
Then again, if you are attempting to explode before you compact, then LGTM